lib/ansible/modules/cloud/amazon/rds_instance.py

```diff
@@ -353,19 +356,28 @@
     'zone': 'AvailabilityZone',
 }
 
+def eprint(*args, **kwargs):
+    print(args, file=sys.stderr, **kwargs)
 
-def await_resource(conn, resource, status, module):
+def await_resource(conn, resource, status, module, await_pending=None):
     wait_timeout = module.params.get('wait_timeout') + time.time()
-    while wait_timeout > time.time() and resource.status != status:
+    # Refresh the resource immediately in case we just changed it's state; should we sleep first?
+    resource = get_db_instance(conn, resource.name)
+    eprint(str(resource))
```

Comment:
I had a look at q; it's fine for debugging and I'll probably use it somehow, but it seems that it could leave debugging information in a public file in /tmp which could be a security problem if it accidentally went into production so I'm not totally convinced by it and I won't use it in any case in code I push up to github until I better understand how to use it securely.